### PR TITLE
(populate) - support (nested) interfaces & unions in the return type

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Fixes https://github.com/FormidableLabs/urql/issues/909

## Summary

Adds support for nested complex types

### CODE REVIEW

File: exchanges/populate/src/populateExchange.ts [hunk 1]
```diff
@@ -138,21 +141,71 @@ export const extractSelectionsFromQuery = (
...
+
+    const validTypeProperties = Object.keys(validTypes);
+
+    selectionSet.selections.forEach(selection => {
+      if (selection.kind === Kind.FIELD) {
+        if (validTypeProperties.includes(selection.name.value)) {
+          if (selection.selectionSet) {
+            selections.push({
+              ...selection,
+              selectionSet: sanitizeSelectionSet(
```
Comment (kadikraman) [inline exchanges/populate/src/populateExchange.ts:159]: Recursion! Fancy.

File: exchanges/populate/src/populateExchange.ts [hunk 2]
```diff
@@ -138,21 +141,71 @@ export const extractSelectionsFromQuery = (
...
+                  t.toString()
+                ),
+              });
+            });
+          } else if (type) {
+            newFragments.push({
+              kind: Kind.FRAGMENT_DEFINITION,
+              typeCondition: {
+                kind: Kind.NAMED_TYPE,
+                name: nameNode(type.toString()),
```
Comment (kadikraman) [inline exchanges/populate/src/populateExchange.ts:203]: Any reason we need to call `.toString()` on type? Isn't it a string already?

Comment (JoviDeCroock) [reply exchanges/populate/src/populateExchange.ts:203]: It's an instance of the TypeNode class, we did this before as well. Could try and explore different avenues

**Review by kadikraman — APPROVED** [review]
LGTM! 👍

**Review by andyrichardson — APPROVED** [review]
Looks like a great addition!

**Pull request merged**